NDTensors/src/dense/dense.jl

```diff
@@ -18,8 +18,24 @@ struct Dense{ElT,VecT<:AbstractArray} <: TensorStorage{ElT}
   function Dense{ElT,VecT}(data::Array) where {ElT,VecT<:AbstractArray{ElT}}
     return new{ElT,VecT}(vec(data))
   end
+
+  function Dense{ElT, VecT}() where {ElT, VecT<:AbstractArray{ElT}}
+    return new{ElT, VecT}(VecT(undef, 0))
+  end
 end
 
+## We need a general typeinfo constructor for Dense
+## This constructor will be called by all other versions
+Dense(eltype::Type, datatype::Type) = default_storage_type(Dense, eltype, datatype)
```

Comment:
These constructors are a bit funny to me, since they are returning types, not objects.

I think these should be functions:
```julia
dense_type(eltype::Type{<:Number}) = adapt_type(eltype, Dense)
dense_type(datatype::Type{<:AbstractArray}) = adapt_type(datatype, Dense)
```
Also I would try to avoid the version that accepts multiple types to keep things simple (though maybe there is a need for it that I'm not seeing right now).